docs/src/lib/transform.ts

```diff
@@ -49,6 +53,28 @@ function transformFunction({
   return { id, name, description, methods, sourceUrl };
 }
 
+function transformSummary(
+  summary: ReadonlyArray<JSONOutput.CommentDisplayPart>,
+  functionNames: Set<string>,
+): string | undefined {
+  if (summary.length === 0) {
+    return;
+  }
+  return summary
+    .map(({ kind, text }) => {
+      if (kind !== "code") {
+        return text;
+      }
+      const codeContent = text.slice(1, -1);
+      // If this is a function name, link to its anchor:
+      if (functionNames.has(codeContent)) {
+        return `[${text}](#${codeContent})`;
+      }
+      return text;
+    })
```

Comment:
Instead of pulling out the whole transformSummary, i'd pull out only this part as it's own utility function transformCommentDisplayPart or something like that, so we can reuse it for other parts if we ever want to.

It's more a pet peeve of mine, but I usually don't like utility functions which just wrap a "map" or "filter" which does most of the logic, in those cases I usually prefer the inner function be pulled out instead.

```suggestion
    .map((part) => transformCommendDisplayPart(part, functionNames))
```


```ts
function transformCommentDisplayPart(
  { kind, text }: JSONOutput.CommentDisplayPart,
  functionNames: ReadonlySet<string>,
): string {
  if (kind !== "code") {
    return text;
  }

  const codeContent = text.slice(1, -1);

  return functionNames.has(codeContent)
    // If this is a function name, link to its anchor:
    ? `[${text}](#${codeContent})`
    : text;
}